db/db_iterator_test.cc

```diff
@@ -3555,6 +3555,96 @@ TEST_F(DBIteratorTest, ErrorWhenReadFile) {
   iter->Reset();
 }
 
+TEST_F(DBIteratorTest, IteratorsConsistentViewTest) {
+  Options options = GetDefaultOptions();
+  CreateAndReopenWithCF({"cf_1", "cf_2"}, options);
+
+  for (int i = 0; i < 3; ++i) {
+    ASSERT_OK(Put(i, "cf" + std::to_string(i) + "_key",
+                  "cf" + std::to_string(i) + "_val"));
+  }
+
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->LoadDependency(
+      {{"DBImpl::BGWorkFlush:done",
+        "DBImpl::MultiCFSnapshot::AfterGetSeqNum1"}});
+
+  bool flushed = false;
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->SetCallBack(
+      "DBImpl::MultiCFSnapshot::AfterRefSV", [&](void* /*arg*/) {
+        if (!flushed) {
+          for (int i = 0; i < 3; ++i) {
+            ASSERT_OK(Put(i, "cf" + std::to_string(i) + "_key",
+                          "cf" + std::to_string(i) + "_val_new"));
+            ASSERT_OK(Flush(i));
+          }
+          flushed = true;
+        }
+      });
+  ROCKSDB_NAMESPACE::SyncPoint::GetInstance()->EnableProcessing();
+  // Explicit snapshot wouldn't force reloading all svs. We should expect old
+  // values
+  {
+    const Snapshot* snapshot = db_->GetSnapshot();
+    ReadOptions read_options;
+    read_options.snapshot = snapshot;
+    std::vector<Iterator*> iters;
+    ASSERT_OK(db_->NewIterators(read_options, handles_, &iters));
+
+    for (int i = 0; i < 3; ++i) {
+      auto iter = iters[i];
+      ASSERT_OK(iter->status());
+      iter->SeekToFirst();
+      ASSERT_EQ(IterStatus(iter), "cf" + std::to_string(i) + "_key->cf" +
```

Comment:
Will this get "_val_new" for cf_1 and cf_2 without the `MultiCFSnapshot` change? Just trying to understand what inconsistency symptom this is like before the fix.

Reply:
I ended up separating the two test cases implicit/explicit snapshot.